### ISSUE TITLE
GUI statusline colors are bleeding out

### ISSUE THREAD

**Issue by equalsraf** [equalsraf/neovim-qt#11]
The theme is wombat256, just open a file and scroll down.

![qtbug2](https://cloud.githubusercontent.com/assets/715238/7297070/373e1ee8-e9bd-11e4-901c-577723238c1a.png)

 We are not setting (or resetting) the bg colors properly, it also seems to apply to font attributes being missed

![qtbug3](https://cloud.githubusercontent.com/assets/715238/7297125/e94ab8bc-e9bd-11e4-9da0-8ec9c1244fb7.png)

e.g. this diff

```diff
diff --git a/src/gui/shell.cpp b/src/gui/shell.cpp
index 83942b9..b0ccaf3 100644
--- a/src/gui/shell.cpp
+++ b/src/gui/shell.cpp
@@ -305,6 +305,7 @@ void Shell::handleScroll(const QVariantList& args, QPainter& painter)
 /** Ready a painter with Neovim settings */
 void Shell::setupPainter(QPainter& painter)
 {
+       m_hg_background = m_background;
        painter.setFont(m_font);
        painter.setPen(m_hg_foreground);
        painter.setBackground(m_hg_background);

```

helps a bit with a background, but not completely, it also causes other issues

![qtbug4](https://cloud.githubusercontent.com/assets/715238/7297149/2d5d5dfc-e9be-11e4-8d17-174299c03498.png)

TLDR; we are setting or unsetting the drawing properties when we shouldn't.

**Comment by equalsraf** [comment]
This seems to fix things for the background color when scrolling, we were using the highlight background to paint on scroll.

 ```diff
diff --git a/src/gui/shell.cpp b/src/gui/shell.cpp
index 83942b9..10aef78 100644
--- a/src/gui/shell.cpp
+++ b/src/gui/shell.cpp
@@ -298,7 +298,8 @@ void Shell::handleScroll(const QVariantList& args, QPainter& painter)
 
        QImage copy = m_image.copy(rect);
        painter.drawImage(pos, copy);
-       painter.eraseRect(exposed);
+       // Scroll always uses the background color, not the highlight
+       painter.fillRect(exposed, m_background);
        update(m_scroll_region);
 }
 ```

**Comment by equalsraf** [comment]
Fixed in cf2d2c3d510501d6d1c653274e7d75bf5ed6c836
